tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -155,44 +136,146 @@ var _ = Describe("odo devfile push command tests", func() {
 				podName,
+				"",
 				namespace,
-				[]string{"stat", "/projects/server.js"},
+				[]string{"stat", "/projects/nodejs-web-app/app/app.js"},
 				func(cmdOp string, err error) bool {
 					statErr = err
 					return true
 				},
 			)
 			Expect(statErr).To(HaveOccurred())
-			Expect(statErr.Error()).To(ContainSubstring("cannot stat '/projects/server.js': No such file or directory"))
+			Expect(statErr.Error()).To(ContainSubstring("cannot stat '/projects/nodejs-web-app/app/app.js': No such file or directory"))
 		})
 
 		It("should build when no changes are detected in the directory and force flag is enabled", func() {
-			// Devfile push requires experimental mode to be set
-			helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
 
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs-multicontainer"), context)
-			cmpName := helper.RandString(6)
-			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, "--project", namespace, cmpName)
 			helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--namespace", namespace)
 
 			// use the force build flag and push
 			output := helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--namespace", namespace, "-f")
 			Expect(output).To(Not(ContainSubstring("No file changes detected, skipping build")))
 		})
 
+		It("should execute the default devbuild and devrun commands if present", func() {
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
+
+			output := helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--namespace", namespace)
+			Expect(output).To(ContainSubstring("Executing devbuild command npm install"))
+			Expect(output).To(ContainSubstring("Executing devrun command nodemon app.js"))
+		})
+
+		It("should be able to handle a missing devbuild command", func() {
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
+			helper.RenameFile("devfile.yaml", "devfile-old.yaml")
+			helper.RenameFile("devfile-without-devbuild.yaml", "devfile.yaml")
+
+			output := helper.CmdShouldPass("odo", "push", "--devfile", "devfile.yaml", "--namespace", namespace)
+			Expect(output).NotTo(ContainSubstring("Executing devbuild command"))
+			Expect(output).To(ContainSubstring("Executing devrun command npm install && nodemon app.js"))
+		})
+
+		It("should error out on a missing devrun command", func() {
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs"), context)
+
+			// Rename the devrun command
+			helper.ReplaceString(filepath.Join(context, "devfile.yaml"), "devrun", "randomcommand")
+
+			output := helper.CmdShouldFail("odo", "push", "--devfile", "devfile.yaml", "--namespace", namespace)
+			Expect(output).NotTo(ContainSubstring("Executing devrun command"))
+			Expect(output).To(ContainSubstring("Failed to start component with name"))
```

Comment:
You can probably remove some of the output checks in the test cases you added, such as the middle one here.

I.e. it's probably sufficient to just check that `Executing devrun command` doesn't exist and that `The command \"devrun\" was not found in the devfile` was found.